src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,39 @@
+package core.basesyntax;
+
+public class FigureSupplier {
+    private ColorSupplier colorSupplier;
+
+    public FigureSupplier(ColorSupplier colorSupplier) {
+        this.colorSupplier = colorSupplier;
+    }
+
+    public Figure getRandomFigure() {
+        switch ((int) (Math.random() * 6)) {
```

Comment:
6 is a magic number)) let's use constant instead - https://mate-academy.github.io/style-guides/java/java.html#s5.2.4-constant-names